### ISSUE TITLE
Possibly improve freetext search by using Mango Index instead of view

### ISSUE THREAD

**Issue by jkuester** [medic/cht-core#8998]
**What feature do you want to improve?**
[This blog post](https://reintech.io/blog/efficient-full-text-search-couchdb-mango-queries) notes that [Mango queries](https://pouchdb.com/guides/mango-queries.html) (aka [`_find` requests](https://docs.couchdb.org/en/stable/api/database/find.html)) can be leveraged with pre-built indexes to provide performant full-text search.  

It might be worth investigating if this is true and if it can provide better performance/functionality than our current "freetext" search approach (which is very brute-force TBH).  

**Describe the improvement you'd like**

Currently doing freetext searches (e.g. searching for a contact by name) requires some [nasty views](https://github.com/medic/cht-core/blob/dac993b4529cd70fea76cf9678f0c2e02705e7e5/ddocs/medic-db/medic-client/views/contacts_by_freetext/map.js#L7) that do not index words `< 3` characters.  See https://github.com/medic/cht-core/issues/8832 for the UX challenges this poses to users. 

A more functional full-text search would be useful to users and perhaps Mango indexes could get us that without sacrificing much in terms of performance, complexity, or DB index/view size.  The [`_find` docs](https://docs.couchdb.org/en/stable/api/database/find.html#db-index) note: 

> Mango wraps several index types, starting with the Primary Index out-of-the-box. Mango indexes, with index type json, are built using MapReduce Views.

I expect this means that Mango indexes on the Couch server leverage Erlang views, which [could be more performant](https://github.com/medic/cht-core/issues/8835).  But we would need to measure the performance impacts for both the server and the offline Pouch implementation (since that will not be built on Erlang).

**Additional context**
@ChinHairSaintClair this might be of interest to you since it touches on things we discussed [in this thread](https://forum.communityhealthtoolkit.org/t/mitigate-duplicate-data-capture/3313)!

**Comment by jkuester** [comment]
Another thing that just struck me is that a Mango index that is implemented with Erlang views on the server and JS views on Pouch is basically the (already implemented) abstraction layer that @nydr and I were dreaming about awhile back (where you could define your query once and have it automatically interpolated into different view structures on the client vs server)....

**Comment by dianabarsan** [comment]
@jkuester do you still want this around?

**Comment by jkuester** [comment]
No, this is now irrelevant after https://github.com/medic/cht-core/commit/f1bdfc07c0f4b1170e0d8be671cd27aa105baff4
